hail/src/main/scala/is/hail/types/physical/stypes/concrete/SUnreachable.scala

```diff
@@ -14,7 +14,7 @@ object SUnreachable {
   def fromVirtualType(t: Type): SType = {
     require(t.isRealizable)
     t match {
-      case t if t.isPrimitive => SType.canonical(t)
+      case Primitive(t) => SType.canonical(t)
```

Comment:
Unapply methods can be powerful, but I'm wary of overusing them. They create the illusion of a member of the sum type that doesn't really exist, which adds some cognitive overhead and potential for confusion. There should be sufficient benefit to justify that complexity. Is `Primitive` adding anything more than saving a few characters in case statements? (Genuine question; I'm open to being convinced!)

Reply:
stylistic. I'll remove.